src/datadog/trace_segment.cpp

```diff
@@ -337,5 +363,48 @@ void TraceSegment::inject(DictWriter& writer, const SpanData& span) {
   }
 }
 
+Expected<void> TraceSegment::extract(const DictReader& headers) {
+  if (!awaiting_delegated_sampling_decision_) return {};
+
+  if (auto sampling_delegation_response =
+          headers.lookup(DD_SAMPLING_DELEGATION_RESPONSE_HEADER)) {
+    auto sampling_decision =
+        parse_sampling_delegation_response(*sampling_delegation_response);
+    if (!sampling_decision) {
+      return sampling_decision.error();
+    }
+
+    // std::swap(sampling_decision_, *sampling_decision);
+    sampling_decision_ = *sampling_decision;
+    awaiting_delegated_sampling_decision_ = false;
+
+    if (spans_.front()->parent_id == 0) {
```

Comment:
Maybe a comment here.

Reply:
Resolved in 4c20da00573f307d44b3554528baf3c22727da06